libs/designsystem/shared/floating/src/floating.directive.ts

```diff
@@ -402,6 +402,7 @@ export class FloatingDirective implements OnInit, OnDestroy {
   }
 
   private handleClickOutsideHostElement(event: Event): void {
+    if (!(event.target instanceof HTMLElement)) return;
```

Comment:
Why this guard? What can be clicked that's not an `HTMLElement`? 🤔

Reply:
event.target is has `Node` type, so I figured this early return was the easiest way to make typechecking behave on the next line:
```
const clickedOnReferenceWithClickTriggerEnabled: boolean =
      this.reference?.nativeElement.contains(event.target)
```

Reply:
Definitely open to suggestions for a more slick version of this!

Reply:
As `this.reference?.nativeElement.contains` takes a `Node` I don't see a reason for this guard / type check...

Reply:
Whoops, I edited my original comment now meant to say `EventTarget` not `Node`